File: src/test/java/org/appsugar/archetypes/repository/UserRepositoryTest.java
package org.appsugar.archetypes.repository;

import lombok.extern.slf4j.Slf4j;
import org.appsugar.archetypes.domain.User;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.List;

/**
 * 用户数据资源测试类
 */
@Slf4j
public class UserRepositoryTest extends BaseRepositoryTest {
    @Autowired
    UserRepository userRepository;


    @Test
    public void testFindAll() {
        List<User> users = userRepository.findAll();
        log.debug("testFindAll  result is {}", users);
    }

}

File: src/test/kotlin/org/appsugar/archetypes/repository/UserRepositoryTest.kt
package org.appsugar.archetypes.repository

import kotlinx.coroutines.future.await
import kotlinx.coroutines.runBlocking
import org.appsugar.archetypes.BaseTestCase
import org.appsugar.archetypes.repository.jpa.UserCondition
import org.appsugar.archetypes.repository.jpa.UserJpaRepository
import org.appsugar.archetypes.repository.jpa.toPredicate
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired

class UserRepositoryTest : BaseTestCase() {
    @Autowired
    lateinit var repository: UserJpaRepository

    @Test
    fun testFindByLoginName() = runBlocking {
        val loginName = "admin"
        val user = repository.findByLoginName(loginName).await()
        logger.debug("find user by loginName $loginName  result is $user roles is ${user?.roles}  ${repository.findById(-1)}")
        Assertions.assertNotNull(user)
    }

    @Test
    fun testFindByCondition() {
        val condition = UserCondition(name = "NewYoung", loginName = "123")
        val users = repository.findAll(condition.toPredicate())
        logger.debug("testFindByCondition {} result is {} ", condition, users)
    }
}


